react/components/molecules/search/search.js

```diff
@@ -0,0 +1,155 @@
+import React, { PureComponent } from "react";
+import { Animated, StyleSheet, View, ViewPropTypes } from "react-native";
+import PropTypes from "prop-types";
+
+import { ButtonIcon, TextArea } from "../../atoms";
+
+export class Search extends PureComponent {
+    static get propTypes() {
+        return {
+            value: PropTypes.string,
+            placeholder: PropTypes.string,
+            buttonProps: PropTypes.object,
+            onFocus: PropTypes.func,
+            onValue: PropTypes.func,
+            onBlur: PropTypes.func,
+            onClear: PropTypes.func,
+            onSubmit: PropTypes.func,
+            style: ViewPropTypes.style
+        };
+    }
+
+    static get defaultProps() {
+        return {
+            value: undefined,
+            placeholder: "Search",
+            buttonProps: {},
+            onFocus: () => {},
+            onValue: value => {},
+            onBlur: () => {},
+            onClear: () => {},
+            onSubmit: () => {},
+            style: {}
+        };
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            value: this.props.value,

```

Comment:
Even though this brings no problem, in the majority of the components in react we are using `valueData`, similar to vue. This helps with code readability, since its very ease to confuse prop "value" it state "value"